zerver/forms.py

```diff
@@ -69,8 +69,9 @@ class RegistrationForm(forms.Form):
     # actually required for a realm
     password = forms.CharField(widget=forms.PasswordInput, max_length=100,
                                required=False)
-    realm_name = forms.CharField(max_length=100, required=False)
-
+    if settings.OPEN_REALM_CREATION:
+        realm_name = forms.CharField(max_length=100, required=False)
+        realm_subdomain = forms.CharField(max_length=40, required=False)
```

Comment:
It's worth adding a comment here though explaining why these aren't required.

Also, I think we should have these present unconditionally, since even with `OPEN_REALM_CREATION` off, one can still create realms via the `generate_realm_creation_link` code path.